main.go

```diff
@@ -2,8 +2,63 @@ package main
 
 import (
 	"fmt"
+	"log"
+
+	"github.com/sensu-community/sensu-plugin-sdk/sensu"
+	"github.com/sensu/sensu-go/types"
+)
+
+// Plugin ...
+type Plugin struct {
+	sensu.PluginConfig
+	Example string
+}
+
+// Options ...
+type Options struct {
+	Example sensu.PluginConfigOption
+}
+
+var (
+	plugin = Plugin{
+		PluginConfig: sensu.PluginConfig{
+			Name:     "{{ .GithubProject }}",
+			Short:    "{{ .Description }}",
+			Timeout:  10,
+			Keyspace: "sensu.io/plugins/{{ .GithubProject }}/config",
+		},
+	}
+
+	checkOptions = Options{
+		Example: sensu.PluginConfigOption{
+			Path:      "example",
+			Env:       "CHECK_EXAMPLE",
+			Argument:  "example",
+			Shorthand: "e",
+			Default:   "",
+			Usage:     "An example configuration option",
+			Value:     &plugin.Example,
+		},
+	}
+
+	options = []*sensu.PluginConfigOption{
+		&checkOptions.Example,
+	}
 )
 
 func main() {
-	fmt.Println("Hello, World!")
+	check := sensu.NewGoCheck(&plugin.PluginConfig, options, checkArgs, executeCheck, false)
+	check.Execute()
+}
+
+func checkArgs(event *types.Event) (int, error) {
+	if len(plugin.Example) == 0 {
+		return 0, fmt.Errorf("--example or CHECK_EXAMPLE environment variable is required")
```

Comment:
The first return value of `checkArgs()` is the status code that the check should exit with. I don't think it should be `0` in case of an error! See the [constants defined in the SDK](https://github.com/sensu-community/sensu-plugin-sdk/blob/master/sensu/gocheck.go#L11-L16) for options.